src/core/particle_data.cpp

```diff
@@ -64,12 +65,9 @@
  * variables
  ************************************************/
 // List of particles for grandcanonical simulations
-TypeOfIndex Type;
-IndexOfType Index;
-TypeList *type_array;
 int number_of_type_lists;
-int GC_init;
-int Type_array_init;
+bool GC_init;
+std::unordered_map<int, std::unordered_set<int>> ParticleTypeMap{};
```

Comment:
Please adhere to the naming convention. Variable names should be lower case and not camel.